### PR TITLE
Dialog: minimize/maximize functionality

### PR DESCRIPTION
**Pull request by AlexanderGeere**
## Description

A function allow a dialog to be minimized and maximized (collapsed into the header)

## GitHub Issue
[#1496](https://github.com/GEOLYTIX/xyz/issues/1496)

## Type of Change
- ✅ New feature (non-breaking change which adds functionality)
- ✅ Documentation
- ✅ Testing

## How have you tested this? 
Tested locally using the test suite.

## Testing Checklist 
- ✅ Updated Existing Tests
- ✅ New Tests Added
- ✅ Ran locally on my machine

## Code Quality Checklist
- ✅ My code follows the guidelines of XYZ
- ✅ My code has been commented
- ✅ Documentation has been updated
- ✅ New and existing unit tests pass locally with my changes
- ✅ Main has been merged into this PR

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208488860427081

### CODE REVIEW

File: public/icons/icon-close.svg
Comment (RobAndrewHurst) [inline public/icons/icon-close.svg:1]: We changed the close icon in the core as it was weirdly formatted on the canvas!

**Review by dbauszus-glx — APPROVED** [review]
Please check my commit in regards to the minimize logic.

This should be handled with a one line rather than complex add/remove classes.

```js
 dialog.minimizeBtn &&= mapp.utils.html`<button
    class="mask-icon minimize-btn"
    onclick=${e => {
      e.target.closest('dialog').classList.toggle('minimized')
    }}>`
```

All the styling is controlled by css.
```css
dialog.minimized .content {
  display: none;
}

dialog .minimize-btn.mask-icon {
  -webkit-mask-image: url("../icons/icon-fullscreen.svg");
  mask-image: url("../icons/icon-fullscreen.svg");
}

dialog.minimized .minimize-btn.mask-icon {
  -webkit-mask-image: url("../icons/icon-remove.svg");
  mask-image: url("../icons/icon-remove.svg");
}
```

I moved the dialog css into it's own file. I can see that the style rules are a bit all over the place but I can't get myself to order and review them all at this stage.

I removed a couple of data-id where not required. The data-id is a fallback where classes are not available or where we want to provide a custom data-id to an element. But selection of a header in a dialog can be done with `dialog header` css specificity.

I ordered button and dialog assignment in the method and changed header to headerHTML. The header provided as param is not the header used in the html template.

You might otherwise get issues if you re-create the dialog with the new flag as you nest the content in itself.

```js
dialog.headerHtml = mapp.utils.html`<header
    class=${dialog.headerDrag ? 'headerDrag': ''}>
    ${dialog.header}${dialog.minimizeBtn}${dialog.closeBtn}`
```

I removed a couple of unnecessary html.node tag methods in favour of html tag methods. The dialog itself is turned into a node once all the html template tags are put together.

I removed some nesting.
Screenshot from 2024-10-08 16-40-56

Instead of header > header > dialog-controls div > button I changed this to header > button.

image

**Review by RobAndrewHurst — APPROVED** [review]
@dbauszus-glx Nice one!

I noticed that the minimize button icon was the full screen icon, and it should be the 'remove' icon.

image

I also updated the tests to work again.

Please make sure that if you make changes that the local tests pass :)

I am happy to approve this!

**Review by cityremade — CHANGES_REQUESTED** [review]
[
Screenshot 2024-10-09 at 09 42 21
](url)

shouldn't it sit in the top corner? at the moment it's under the header for me 🤔

**Comment by RobAndrewHurst** [comment]
> [ Screenshot 2024-10-09 at 09 42 21 ](url)
> 
> shouldn't it sit in the top corner? at the moment it's under the header for me 🤔

Can you please write a test for this?

**Comment by dbauszus-glx** [comment]
https://github.com/GEOLYTIX/xyz/pull/1541/commits/b805468114359bd04873e13cea60039c2190ada7

The confirm dialog will delete minimizeBtn and closeBtn property flags as these are not shown on confirm dialogs.

CSS has been updated. There shouldn't be a rule for <div> inside the alert-confirm class.

The alert confirm dialog is either modal or dialog class. Border and background should be assigned to these classes.

There should not be multiple box-shadow definitions in CSS. These are at best dusplicates and in the worst case completely different definitions for the same thing. There is a box-shadow class which should be assigned for the box-shadow css rule.

Moved modal class to be aftter the dialog class in _dialog.css sheet. As mentioned previously. There is probably still a lot of stuff in that sheet which hasn't been reviewed proper.

**Review by simon-leech — APPROVED** [review]

**Pull request merged**